lib/puppet/parser/functions/consul_sorted_json.rb

```diff
@@ -1,43 +1,91 @@
 require 'json'
 
-def sorted_json(obj)
-  case obj
+module ConsulSortedJson
+
+  module_function
+
+
+  def sorted_json(config_hash, pretty)
+    cleaned = convert_integers( config_hash )
+    sorted  = sort_keys( cleaned )
+
+    if pretty
+      JSON.pretty_generate( sorted )
```

Comment:
Its not the same. If you notice, in the currently implementation we only do `to_json` on **non-array** and **non-hash** objects, for arrays and hashes we do string concatenation to ensure order, and once we sort we do *zero* post-processing. But here you are processing the entire hash, which will *not* guarantee order :smile: